packages/twenty-front/src/modules/object-record/record-table/hooks/useCreateNewTableRecords.ts

```diff
@@ -54,30 +57,69 @@ export const useCreateNewTableRecord = ({
     shouldMatchRootQueryFilter: true,
   });
 
-  const createNewTableRecord = async () => {
-    const recordId = v4();
+  const navigate = useNavigate();
 
-    if (isCommandMenuV2Enabled) {
-      await createOneRecord({ id: recordId });
+  const createNewTableRecord = useRecoilCallback(
+    ({ set }) =>
+      async () => {
+        const recordId = v4();
 
-      openRecordInCommandMenu(recordId, objectMetadataItem.nameSingular);
-      return;
-    }
+        if (isCommandMenuV2Enabled) {
+          // TODO: Generalize this behaviour, there will be a view setting to specify
+          // if the new record should be displayed in the side panel or on the record page
+          if (
+            objectMetadataItem.nameSingular === CoreObjectNameSingular.Workflow
```

Comment:
ok but we could introduce a hook or an util to encapsulate this shouldRedirectToShowPageOnCreation logic